src/main/java/dialog/TaskDialog.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * @author Kan Jitpakdi
+ * @author GitHub: kanjitp
+ * @version 0.01
+ * @since 0.00
+ */
+public class TaskDialog extends Dialog {
+    // This class is a child of dialog class which allow the user to interact with the task
+    private TaskList taskList;
+
+    /** private constructor only to be used within the class */
+    private TaskDialog(ArrayList<String> sentences, TaskList taskList) {
+        super(sentences);
+        this.taskList = taskList;
+    }
+
+    /**
+     * Factory method for creating TaskDialog as Dialog type (to be cast by developer)
+     * Factory method from TaskDialog allow the user to fetch the existing Dialog in the archive.
+     * The default TaskList will have an empty ArrayList of task.
+     *
+     * @param id the id of the TaskDialog
+     * @return TaskDialog of type Dialog
+     */
+    public static Dialog generate(String id) {
+        if (archive.containsKey(id)) {
+            return archive.get(id);
+        } else {
+            final Dialog newDialog = new TaskDialog(new ArrayList<>(), new TaskList(new ArrayList<>()));
+            archive.put(id, newDialog);
+            return newDialog;
+        }
+    }
+
+    /**
+     * Factory method for creating TaskDialog as Dialog type (to be cast by developer)
+     * Factory method from TaskDialog allow the user to fetch the existing Dialog in the archive.
+     *
+     * @param id       the id of the TaskDialog
+     * @param taskList the taskList the TaskDialog is going to have
+     * @return TaskDialog of type Dialog
+     */
+    public static Dialog generate(String id, TaskList taskList) {
+        if (archive.containsKey(id)) {
+            return archive.get(id);
+        } else {
+            final Dialog newDialog = new TaskDialog(new ArrayList<>(), taskList);
+            archive.put(id, newDialog);
+            return newDialog;
+        }
+    }
+
+    /**
+     * Add the task to the TaskList of this Dialog and print out feedback to the user
+     *
+     * @param task the task to be added to the TaskList of this TaskDialog
+     * @throws DialogException dialog cannot have the same id while the app is running
+     */
+    public void addTask(Task task) throws DialogException {
+        Dialog addDialog = Dialog.generate(Integer.toString(task.hashCode()));
+        taskList.add(task);
+        addDialog.add("Got it. I've added this task:");
+        addDialog.add("  " + task);
+        addDialog.add("Now you have " + this.taskList.length() + " tasks in the list.");
+        System.out.println(addDialog);
+    }
+
+    /**
+     * Return TaskDialog filtered by the specific deadline of format yyyy-MM-dd
+     *
+     * @param deadline the deadline of format yyyy-MM-dd
+     * @return TaskDialog with new TaskList stored in it
+     * @throws InvalidTimeFormatException time of illegal format (not yyyy-MM-dd)
+     */
+    public TaskDialog by(String deadline) throws InvalidTimeFormatException {
```

Comment:
Do you want to rename this method to be a verb? e.g. getFromDeadline?

Reply:
Yeah, most likely a good idea I was at that time thinking of stream chaining so I named it in such manner